File: android/src/main/java/com/schoolbell_e/plugins/mlkit_language/MLKitLanguagePlugin.java
package com.schoolbell_e.plugins.mlkit_language;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "MLKitLanguage")
public class MLKitLanguagePlugin extends Plugin {

    private MLKitLanguage implementation = new MLKitLanguage();

    @PluginMethod()
    public void identifyLanguage(PluginCall call) {
        String value = call.getString("text");
        implementation.identifyLanguage(value, new MLKitLanguage.Callback() {
            @Override
            public void resolve(String lang) {
                JSObject ret = new JSObject();
                ret.put("languageCode", lang);
                call.resolve(ret);
            }

            @Override
            public void reject(String errorMessage) {
                call.reject(errorMessage);
            }
        });
    }
}

File: ios/Plugin/MLKitLanguagePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(MLKitLanguagePlugin)
public class MLKitLanguagePlugin: CAPPlugin {
    private let implementation = MLKitLanguage()

    @objc func identifyLanguage(_ call: CAPPluginCall) {
        let value = call.getString("text") ?? ""
        if value.isEmpty {
            call.reject("empty string given")
            return
        }
        
        let lang = implementation.identifyLanguage(for:value)
        call.resolve([
            "languageCode": lang
        ])
    }
}
